packages/twenty-front/src/modules/object-record/record-group/states/recordGroupFieldMetadataComponentState.ts

```diff
@@ -0,0 +1,11 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { createComponentStateV2 } from '@/ui/utilities/state/component-state/utils/createComponentStateV2';
+import { ViewComponentInstanceContext } from '@/views/states/contexts/ViewComponentInstanceContext';
+
+export const recordGroupFieldMetadataComponentState = createComponentStateV2<
+  FieldMetadataItem | undefined
+>({
+  key: 'recordGroupFieldMetadataComponentState',
+  defaultValue: undefined,
+  componentInstanceContext: ViewComponentInstanceContext,
```

Comment:
should not use ViewComponentInstanceContext in RecordGroups

Reply:
The context war haha, actually mixing context between different modules can be tricky because some selectors or are mixing some states from different modules.
It's a bit due to the refactor, but also due to the structure of record-group, maybe we should move it inside record-index.
Mixing states in selectors can actually work as the instanceId between context is often the same, but it's not a good option.

Reply:
yes, ok, let's do that in a folllow up